gobject-introspection wants to use darwin host otool instead of our stdenv one

It then fails on machines without command-line tools or Xcode installed.

Yep, I think I've bumped into the same thing trying to install `nixpkgs.pass`.

```
xcode-select: note: no developer tools were found at '/Applications/Xcode.app', requesting install. Choose an option in the dialog to download the command line developer tools.
ERROR: can't resolve libraries to shared libraries: gobject-2.0, glib-2.0
Makefile:3429: recipe for target 'GLib-2.0.gir' failed
make[2]: *** [GLib-2.0.gir] Error 1
make[2]: Leaving directory '/private/var/folders/5g/3kx3mc191y93shpp91bg7jyh0000gn/T/nix-build-gobject-introspection-1.46.0.drv-0/gobject-introspection-1.46.0'
Makefile:2720: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/private/var/folders/5g/3kx3mc191y93shpp91bg7jyh0000gn/T/nix-build-gobject-introspection-1.46.0.drv-0/gobject-introspection-1.46.0'
Makefile:1499: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/w904wb32rvfhq08y1x1azz67328nb36f-gobject-introspection-1.46.0.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/a7ib57rywq7ksgcqnfd1xkn6ibflfypw-atk-2.18.0.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/gjrgylvwd30ljyfwflzxyd4a9k0ykqdw-gdk-pixbuf-2.32.3.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/zb1360afrvc16ccvf0a6nlgpixs54z1y-pango-1.38.1.drv’: 1 dependencies couldn't be built
building path(s) ‘/nix/store/r8n2rcwv0fysnc1jzs5v6xsvngpnhmvr-password-store-1.6.5.tar.xz’
cannot build derivation ‘/nix/store/c7c9yiiklp0rlgymk2vkh84yxvjcf866-gtk+-2.24.29.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/ak3bf232r7scbfq8m61whyf0lvm5swwr-pinentry-0.9.6.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/i1madbkmv202rzrsb5qha7yw2aqm3qrf-gnupg-2.0.29.drv’: 1 dependencies couldn't be built
killing process 46716
cannot build derivation ‘/nix/store/5msn2c2cb21zmja8b1nbv3wlw6pi2i5r-password-store-1.6.5.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/5msn2c2cb21zmja8b1nbv3wlw6pi2i5r-password-store-1.6.5.drv’ failed
```

--- a/pkgs/development/libraries/gobject-introspection/default.nix
+++ b/pkgs/development/libraries/gobject-introspection/default.nix
@@ -1,5 +1,5 @@
 { stdenv, fetchurl, glib, flex, bison, pkgconfig, libffi, python
-, libintlOrEmpty, autoconf, automake, otool
+, libintlOrEmpty, cctools
 , substituteAll, nixStoreDir ? builtins.storeDir
 }:
 # now that gobjectIntrospection creates large .gir files (eg gtk3 case)
@@ -20,11 +20,11 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ flex bison pkgconfig python ]
     ++ libintlOrEmpty
-    ++ stdenv.lib.optional stdenv.isDarwin otool;
+    ++ stdenv.lib.optional stdenv.isDarwin cctools;
   propagatedBuildInputs = [ libffi glib ];
 
-  # Tests depend on cairo, which is undesirable (it pulls in lots of
-  # other dependencies).
+  # The '--disable-tests' flag is no longer recognized, so can be safely removed
+  # next time this package changes.
   configureFlags = [ "--disable-tests" ];
 
   preConfigure = ''
@@ -45,7 +45,6 @@ stdenv.mkDerivation rec {
     homepage    = http://live.gnome.org/GObjectIntrospection;
     maintainers = with maintainers; [ lovek323 urkud lethalman ];
     platforms   = platforms.unix;
-    broken      = stdenv.isDarwin;
 
     longDescription = ''
       GObject introspection is a middleware layer between C libraries (using

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -6934,6 +6934,7 @@ let
 
   gobjectIntrospection = callPackage ../development/libraries/gobject-introspection {
     nixStoreDir = config.nix.storeDir or builtins.storeDir;
+    inherit (darwin) cctools;
   };
 
   goocanvas = callPackage ../development/libraries/goocanvas { };

